contrib/internal/httptrace/before_handle.go

```diff
@@ -61,7 +62,10 @@ func BeforeHandle(cfg *ServeConfig, w http.ResponseWriter, r *http.Request) (htt
 	rw, ddrw := wrapResponseWriter(w)
 	rt := r.WithContext(ctx)
 	closeSpan := func() {
-		FinishRequestSpan(span, ddrw.status, cfg.isStatusError, cfg.FinishOpts...)
+		if cfg.IsStatusError != nil && cfg.IsStatusError(ddrw.status) {
+			span.SetTag(ext.Error, fmt.Errorf("%d: %s", ddrw.status, http.StatusText(ddrw.status)))
+		}
+		FinishRequestSpan(span, ddrw.status, cfg.IsStatusError, cfg.FinishOpts...)
```

Comment:
`FinishRequestSpan` already has the [logic](https://github.com/DataDog/dd-trace-go/blob/v1.70.1/contrib/internal/httptrace/httptrace.go#L66) to add error tag, so let's allow `FinishRequestSpan` to perform that functionality (unless there was a specific reason for your change?).

```suggestion
		FinishRequestSpan(span, ddrw.status, cfg.IsStatusError, cfg.FinishOpts...)
```